### DISCUSSION TITLE
switch case brackets

### DISCUSSION THREAD

**Discussion by cterwilliger** [cterwilliger/VeDirectFrameHandler#13] [General]
From email:

I'm using your https://github.com/cterwilliger/VeDirectFrameHandler lib.
Thank you for coding this.

It took some time to understand everything, but there is one question open.

Why did you set brackets in VeDirectFrameHandler.cpp
		case CHECKSUM:
	{
		.......
		break;
	}

As far as I know, in 'switch case' no brackets are needed.

But I tried to delete them and some time later(!!!!) I had huge problems!
I took a long time to find the problem - the missing brackets ;)

So please, can you tell me, what's the secret  :)

Thank you,
andi

**Comment by cterwilliger** [comment]
The VeDirectFrameHandler code is from Victron's VEDirect protocol FAQ: (https://www.victronenergy.com/live/vedirect_protocol:faq)  They wrote that part and I didn't catch it.  Doesn't seem like it should break anything to remove them.  What problems did you see?

**Reply by AndiP-IoT** [reply]
Thank you cterwilliger for placing my email here in the discussion.

> Doesn't seem like it should break anything to remove them.

That's what I thought. So I removed them.

I get the ve.direct text and send it via mqtt to the broker. And I use a I2C display.

One of my victron equipment just send in textmode plain text. In this case everything is fine.
One the other it sends sometimes lines of HEX parameter.
In this case sometime my mqtt stoped publishing.
After some investigation (~2 days) and debug-circling around my code, I realized that the only difference left behind to the original code were the brackets.
I put them in again - everything fine.

BUT!
I just developed my code more and so I took an old version (with the problem) for this discussion to show the problem code here.
And magic : the problem is gone. I do not know what happened. No update, no gui change....

Some time ago I had other problems with compiling WIFI and MQTT for MEGA2560 and a (or THE) WIFI shield- WIFI worked, but MQTT did not work.
Then I worked on other code, other projects. After that - magic - the code for WIFI shield and MQTT worked on that MEGA (without changing something).

So??  For me, sometimes it looks like the compiling (precompiled code?) stucks. And after other code (lib) compiling it works again.
I know, it can't be - but?

I think, my question is closed.
There is no need for these brackets.

Thank you and sorry for the rumor ;)
